src/packages/documents/documents/components/input-document-picker-root/input-document-picker-root.element.ts

```diff
@@ -1,101 +1,268 @@
 
 @customElement('umb-input-document-picker-root')
 export class UmbInputDocumentPickerRootElement extends FormControlMixin(UmbLitElement) {
-	public get unique(): string | null | undefined {
-		return this.#documentPickerContext.getSelection()[0];
-	}
-	public set unique(unique: string | null | undefined) {
-		const selection = unique ? [unique] : [];
-		this.#documentPickerContext.setSelection(selection);
+	protected getFormElement() {
+		return undefined;
 	}
 
-	@property()
-	public set value(unique: string) {
-		this.unique = unique;
-	}
+	@state()
+	private _originManifests: Array<ManifestDynamicRootOrigin> = [];
 
 	@state()
-	private _items?: Array<UmbDocumentItemModel>;
+	private _queryStepManifests: Array<ManifestDynamicRootQueryStep> = [];
+
+	@property({ attribute: false })
+	data?: UmbTreePickerDynamicRoot | undefined;
 
-	#documentPickerContext = new UmbDocumentPickerContext(this);
+	#dynamicRootOrigin?: { label: string; icon: string; description?: string };
 
-	// TODO: DynamicRoot - once feature implemented, wire up context and picker UI. [LK]
-	#dynamicRootPickerContext = {
-		openPicker: () => {
-			throw new Error('DynamicRoot picker has not been implemented yet.');
-		},
-	};
+	#modalContext?: UmbModalManagerContext;
+
+	#openModal?: UmbModalContext;
 
 	constructor() {
 		super();
 
-		this.#documentPickerContext.max = 1;
+		this.consumeContext(UMB_MODAL_MANAGER_CONTEXT, (instance) => {
+			this.#modalContext = instance;
+		});
+
+		this.observe(
+			umbExtensionsRegistry.byType('dynamicRootOrigin'),
+			(originManifests: Array<ManifestDynamicRootOrigin>) => {
+				this._originManifests = originManifests;
+			},
+		);
 
-		this.observe(this.#documentPickerContext.selection, (selection) => (super.value = selection.join(',')));
-		this.observe(this.#documentPickerContext.selectedItems, (selectedItems) => (this._items = selectedItems));
+		this.observe(
+			umbExtensionsRegistry.byType('dynamicRootQueryStep'),
+			(queryStepManifests: Array<ManifestDynamicRootQueryStep>) => {
+				this._queryStepManifests = queryStepManifests;
+			},
+		);
 	}
 
-	protected getFormElement() {
-		return undefined;
+	connectedCallback(): void {
+		super.connectedCallback();
+
+		this.#updateDynamicRootOrigin(this.data);
+		this.#updateDynamicRootQuerySteps(this.data?.querySteps);
+	}
+
+	#sorter?: UmbSorterController<UmbTreePickerDynamicRootQueryStep>;
+
+	#initSorter() {
+		this.#sorter = new UmbSorterController<UmbTreePickerDynamicRootQueryStep>(this, {
+			getUniqueOfElement: (element) => {
+				return element.id;
+			},
+			getUniqueOfModel: (modelEntry) => {
+				return modelEntry.unique;
+			},
+			identifier: 'Umb.SorterIdentifier.InputDocumentPickerRoot',
+			itemSelector: 'uui-ref-node',
+			containerSelector: '#query-steps',
+			onChange: ({ model }) => {
+				if (this.data && this.data.querySteps) {
+					//const steps = [...this.data.querySteps];
+					const querySteps = model; //.map((index) => steps[parseInt(index)]);
```

Comment:
What should happen with these outcommented lines?

Reply:
I'll tidy this up. They were left over code from tweaking yesterday's sorter controller updates.